chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -147,6 +147,77 @@ if test "x$MD5_ADMIN" != "x$MD5_LOCAL"; then
 fi
 
 
+# Helper functions
+# ----------------
+
+get_nodes_allow_fail() {
+  # --fail is important here as we rely on the failure to know if we can
+  # communicate with the admin server
+  local curlargs=(--connect-timeout 60 -s --fail \
+                  -L
+                  -H "Accept: application/json" \
+                  --max-time 240)
+
+  [[ $CROWBAR_KEY ]] && curlargs+=(-u "$CROWBAR_KEY" --digest --anyauth)
+
+  curl "${curlargs[@]}" "http://$ADMIN_IP:3000/crowbar/machines/1.0/"
+}
+
+get_node_allow_fail() {
+  # --fail is important here as we rely on the failure for 404
+  local curlargs=(--connect-timeout 60 -s --fail \
+                  -L
+                  -H "Accept: application/json" \
+                  --max-time 240)
+
+  [[ $CROWBAR_KEY ]] && curlargs+=(-u "$CROWBAR_KEY" --digest --anyauth)
```

Comment:
I guess both functions could also be merged, by just allowing to pass ```hostname``` as a parameter, no?